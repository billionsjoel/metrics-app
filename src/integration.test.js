/* eslint-disable no-tabs */
/* eslint-disable no-mixed-spaces-and-tabs */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import store from './redux/configureStore';
import SingleCountry from './components/countries/country';

describe('tests components', () => {
  const countryData = {
    name: 'uganda',
    date: '23/23/32',
    today_confirmed: 121,
    today_deaths: 12,
    today_new_confirmed: 232,
    today_new_deaths: 12,
  };

  test('checks for a single country component', () => {
    const { asFragment } = render(
      <Provider store={store}>
        <SingleCountry country={countryData} />
      </Provider>,
      {
			  wrapper: MemoryRouter,
      },
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('checks the country name to be uganda', () => {
    const { container } = render(
      <Provider store={store}>
        <SingleCountry country={countryData} />
      </Provider>,
      {
			  wrapper: MemoryRouter,
      },
    );
    expect(container.querySelector('.title').textContent).toBe('uganda');
  });
});
