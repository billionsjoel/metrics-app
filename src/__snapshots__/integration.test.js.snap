// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests components checks for a single country component 1`] = `
<DocumentFragment>
  <div
    class="selfCountries d-flex col"
  >
    <div
      class="countryContainer d-flex col"
    >
      <div
        class="title center bold d-flex"
      >
        <p>
          uganda
        </p>
      </div>
      <div
        class="date d-flex"
      >
        <p
          class="bold"
        >
          Date:
        </p>
        <p>
          23/23/32
        </p>
      </div>
      <div
        class="country-stats"
      >
        <div
          class="d-flex"
        >
          <p
            class="bold"
          >
            Total Comfirmed Cases:
          </p>
          <p>
            121
          </p>
        </div>
        <div
          class="d-flex"
        >
          <p
            class="bold"
          >
            Total Deaths:
          </p>
          <p>
            12
          </p>
        </div>
        <div
          class="d-flex"
        >
          <p
            class="bold"
          >
            Comfirmed New Cases:
          </p>
          <p>
            232
          </p>
        </div>
        <div
          class="d-flex"
        >
          <p
            class="bold"
          >
            Deaths Today:
          </p>
          <p>
            12
          </p>
        </div>
        <div
          aria-hidden="true"
          class="source d-flex col"
        >
          <p>
            Source: Johns Hopkins University.
          </p>
          <p>
            API by Narrativa technology from the Data API.
          </p>
          <p>
            Made with ♥️ by
            <br />
            <span>
              <a
                class="credits"
                href="https://github.com/billionsjoel"
              >
                Billions A Joel
              </a>
            </span>
          </p>
        </div>
      </div>
      <div
        class="backButton"
      >
        <i
          aria-hidden="true"
          class="bi bi-arrow-left-circle pointer"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
